app/services/katello/applicability/applicable_content_helper.rb

```diff
@@ -122,20 +123,24 @@ def newest_distinct_installed_packages_query
       end
 
       def applicable_differences
-        consumer_ids = content_facet.send(applicable_units).pluck("#{content_unit_class.table_name}.id")
-        content_ids = fetch_content_ids
+        ActiveRecord::Base.connection.uncached do
+          consumer_ids = content_facet.send(applicable_units).pluck("#{content_unit_class.table_name}.id")
+          STDOUT.puts("applicable_differences consumer_ids sql: #{consumer_ids}")
+          content_ids = fetch_content_ids
+          STDOUT.puts("applicable_differences content_ids: #{content_ids}")
 
-        to_remove = consumer_ids - content_ids
-        to_add = content_ids - consumer_ids
+          to_remove = consumer_ids - content_ids
+          to_add = content_ids - consumer_ids
 
-        [to_add, to_remove]
+          [to_add, to_remove]
+        end
       end
 
       def insert(applicable_ids)
         unless applicable_ids.empty?
           inserts = applicable_ids.map { |applicable_id| "(#{applicable_id.to_i}, #{content_facet.id.to_i})" }
           sql = "INSERT INTO #{content_facet_association_class.table_name} (#{content_unit_association_id}, content_facet_id) VALUES #{inserts.join(', ')}"
-          ActiveRecord::Base.connection.execute(sql)
+          ActiveRecord::Base.connection.exec_insert(sql)
```

Comment:
Not a part of the fix but the reading I've done tends to favor using `exec_query`  over `execute` due to the possibility of manual garbage collection to be performed. Not sure if the postgresql adapter is an exception to that or not. I opted for `exec_insert` since it wraps `exec_query` and we're doing inserts here.